hmcts/pre-api
# S28 2877: MK Playback Endpoint with JWT

### JIRA ticket(s)
- S28-2877


### Change description
- Add: GET `/media-service/vod` endpoint




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Look on mediakind, see there is NOT a content key policy with a name matching your user id. Call endpoint, see one is created. 
- Call endpoint for valid asset, on asset you are trying to watch, see a new locator with a name that matches the user's id. See start time is now, end time is one hours time. Call the endpoint again after one minute. See times update. 
- Check response works for viewing the asset (see steps below) 
- Call endpoint for a recording that the user does not have access to
- Call endpoint for a recording that does not exist
- Call endpoint for a recording that has been marked as deleted
- On MediaKind, turn off `default` endpoint. See when endpoint called, the streaming endpoint is turned back on.  

**How to see if response object values work for playback using MKPlayer (`HLS`):**
1. Select `Hls` option on MKPlayer
2. Select `ClearKey`
3. Copy `hls_url` value from endpoint response and paste in Source URL. 
4. Copy `token` value from endpoint response and set value of License request headers to `{ "Authorization": "Bearer <token>" }`
5. Click Play (Should start to play)

**How to see if response object values work for playback using MKPlayer (`DASH`):**
1. Select `Dash` option on MKPlayer
2. Select `ClearKey`
3. Copy `dash_url` value from endpoint response and paste in Source URL. 
4. Copy `license_url` value from endpoint response and paste in License URL.
5. Copy `token` value from endpoint response and set value of License request headers to `{ "Authorization": "Bearer <token>" }`
6. Click Play (Should start to play)
